include/eve/algo/zip_iterator.hpp

```diff
@@ -83,6 +83,47 @@ namespace eve::algo
       }
     }
 
+    template <typename _Cardinal>
+    auto cardinal_cast(_Cardinal N) const
+    {
+      return zip_iterator<decltype(I{}.cardinal_cast(N)), decltype(Is{}.cardinal_cast(N))...> {
```

Comment:
can't we use some automatic template type deduction here?

Reply:
I don't think so.

`zip_iterator` will refer to this specific instantiation.